lib/ruby_terraform/command_line/option.rb

```diff
@@ -0,0 +1,60 @@
+require 'json'
+
+module RubyTerraform
+  module CommandLine
+    class Option
+      def initialize(opt_key, value, switch_override: nil)
+        assign_option_value(value)
+        @switch = switch_override || switch_from_key(opt_key)
+      end
+
+      def add_to_subcommand(sub)
+        if value.respond_to?(:keys)
+          add_hash_to_subcommand(sub, value)
+        elsif value.respond_to?(:each)
+          add_array_to_subcommand(sub, value)
+        else
+          add_option_to_subcommand(sub)
+        end || sub
+      end
+
+      private
+
+      attr_reader :switch, :value
+
+      def assign_option_value(value)
+        @value = value
```

Comment:
Feels like this should just be inlined into the constructor.

Reply:
I've been trying to get out of the habit of requiring subclasses to call `super` in their initializers.  I prefer that the common code goes in the base class, and then it calls an explicit method that the sub classes can override.  So this was just a way to let the two boolean subclasses provide their own override.